File: src/order-analytics-api/OrderAnalytics.Adapters/DAO/Tables/OrderTable.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace OrderAnalytics.Adapters.DAO.Tables;

[Table("tb_orders")]
public class OrderTable
{
    [Key]
    [Column("id_order")]
    public int Id { get; set; }
    
    [Column("tx_final_price", TypeName = "numeric")]
    public decimal FinalPrice { get; set; }
    
    [ForeignKey(nameof(ClientId))]
    [Column("fk_id_client")]
    public ClientTable Client { get; set; }
    public int ClientId { get; set; }

    [InverseProperty("Order")]
    public List<ProductTable> Products { get; set; } = [];
}

File: src/order-processor-service/src/main/java/com/gabrielaraujo/order_processor_service/adapters/dao/tables/OrderTable.java
package com.gabrielaraujo.order_processor_service.adapters.dao.tables;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.util.List;

@Entity
@Table(name = "tb_orders")
@Builder
@Getter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class OrderTable {
    @Id
    @Column(name = "id_order")
    private int id;

    @Column(name = "tx_final_price")
    private BigDecimal finalPrice;

    @ManyToOne
    @JoinColumn(name = "fk_id_client")
    private ClientTable client;

    @OneToMany
    @JoinColumn(name = "fk_id_order")
    private List<ProductTable> products;
}
